sh/drive-utils.sh

```diff
@@ -0,0 +1,535 @@
+    "${EXTRA_LOG_CLEAR:-:}" 1 1>&2
+
+    printf "%s\n" "${search_response_drive_info}"
+    return 0
+}
+
+###################################################
+# Search for an existing file on gdrive with write permission.
+# Globals: 2 variables, 2 functions
+#   Variables - API_URL, API_VERSION
+#   Functions - _url_encode, _json_value
+# Arguments: 3
+#   ${1} = file name
+#   ${2} = root dir id of file
+#   ${3} = Access Token
+# Result: print file id else blank
+# Reference:
+#   https://developers.google.com/drive/api/v3/search-files
+###################################################
+_check_existing_file() (
+    [ $# -lt 3 ] && printf "Missing arguments\n" && return 1
+    name_check_existing_file="${1##*/}" rootdir_check_existing_file="${2}" token_check_existing_file="${3}"
+    unset query_check_existing_file search_response_check_existing_file id_check_existing_file
+
+    "${EXTRA_LOG:-:}" "justify" "Checking if" " file exists.." "-" 1>&2
+    query_check_existing_file="$(_url_encode "name='${name_check_existing_file}' and '${rootdir_check_existing_file}' in parents and trashed=false and 'me' in writers")"
+
+    search_response_check_existing_file="$(curl --compressed ${CURL_PROGRESS_EXTRA} \
+        -H "Authorization: Bearer ${token_check_existing_file}" \
+        "${API_URL}/drive/${API_VERSION}/files?q=${query_check_existing_file}&fields=files(id,name,mimeType)&supportsAllDrives=true" || :)" && "${CURL_PROGRESS_EXTRA_CLEAR:-:}" 1 1>&2
+    "${EXTRA_LOG_CLEAR:-:}" 1 1>&2
+
+    id_check_existing_file="$(printf "%s\n" "${search_response_check_existing_file}" | _json_value id 1 1)"
+    [ -n "${id_check_existing_file}" ] && printf "%s\n" "${search_response_check_existing_file}"
+    return 0
+)
+
+###################################################
+# Create/Check directory in google drive.
+# Globals: 2 variables, 2 functions
+#   Variables - API_URL, API_VERSION
+#   Functions - _url_encode, _json_value
+# Arguments: 3
+#   ${1} = dir name
+#   ${2} = root dir id of given dir
+#   ${3} = Access Token
+# Result: print folder id
+# Reference:
+#   https://developers.google.com/drive/api/v3/folder
+###################################################
+_create_directory() (
+    [ $# -lt 3 ] && printf "Missing arguments\n" && return 1
+    dirname_create_directory="${1##*/}" rootdir_create_directory="${2}" token_create_directory="${3}"
+    unset query_create_directory search_response_create_directory folder_id_create_directory
+
+    "${EXTRA_LOG:-:}" "justify" "Creating gdrive dir.." "-" 1>&2
+    query_create_directory="$(_url_encode "mimeType='application/vnd.google-apps.folder' and name='${dirname_create_directory}' and trashed=false and '${rootdir_create_directory}' in parents")"
+
+    search_response_create_directory="$(curl --compressed ${CURL_PROGRESS_EXTRA} \
```

Comment:
Codacy found an issue: [Double quote to prevent globbing and word splitting.](https://app.codacy.com/manual/labbots/google-drive-upload/pullRequest?prid=5865418)